File: VB/MySolution.Module.Web/Security/CustomSecurityStrategyComplex.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Security
Imports System
Imports System.Web

Namespace MySolution.Module.Web.Security
	Public Class CustomSecurityStrategyComplex
		Inherits SecurityStrategyComplex

		Protected Overrides Sub InitializeNewUserCore(ByVal objectSpace As IObjectSpace, ByVal user As Object)
			MyBase.InitializeNewUserCore(objectSpace, user)
		End Sub
		Public Sub InitializeNewUser(ByVal objectSpace As IObjectSpace, ByVal user As Object)
			InitializeNewUserCore(objectSpace, user)
		End Sub
		Public Overrides Sub Logoff()
			If HttpContext.Current.Request.Cookies(".AspNet.External") IsNot Nothing Then
				HttpContext.Current.Response.Cookies(".AspNet.External").Expires = Date.Now.AddDays(-1)
			End If
			MyBase.Logoff()
		End Sub
	End Class
End Namespace

File: MySolution.Module.Web/Security/CustomSecurityStrategyComplex.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Security;
using System;
using System.Web;

namespace MySolution.Module.Web.Security {
    public class CustomSecurityStrategyComplex : SecurityStrategyComplex {
        protected override void InitializeNewUserCore(IObjectSpace objectSpace, object user) {
            base.InitializeNewUserCore(objectSpace, user);
        }
        public void InitializeNewUser(IObjectSpace objectSpace, object user) {
            InitializeNewUserCore(objectSpace, user);
        }
        public override void Logoff() {
            if (HttpContext.Current.Request.Cookies[".AspNet.External"] != null) {
                HttpContext.Current.Response.Cookies[".AspNet.External"].Expires = DateTime.Now.AddDays(-1);
            }
            if (HttpContext.Current.Request.Cookies[".AspNet.Cookies"] != null) {
                HttpContext.Current.Response.Cookies[".AspNet.Cookies"].Expires = DateTime.Now.AddDays(-1);
            }
            base.Logoff();
        }
    }
}
